### PR TITLE
cluster color/size/behavior change at various zoom levels

### PR DESCRIPTION
**Pull request by zjxiaohan**
# Overview

The code in this PR explores different changes that can be made to clusters.

Please do not merge this PR. I created this PR for clarification purpose only.

## Relevant Issue(s)

https://app.asana.com/0/1200435319008187/1200436658127799

## Type of change

- [ ] Filtering (AirTable to Mapbox)
- [ ] Layering
- [x] Mapbox Functionality (Must Have)
- [ ] UX/UI
- [ ] Softr Integration
- [ ] Other "Nice to Have" Features

# Self Review Checklist

Perform a self review of your code using the below guidelines.
If you feel your code does not meet the standards, describe why and the reviewer can work with you to address your concerns.

- [x] My code is DRY ("Do not Repeat Yourself"), i.e. my code is modular and not redundant
- [x] I have commented my code where useful
- [x] The variables in my code are named clearly and convey intent
- [x] There are no obvious optimizations I could make that would improve performance
- [x] I have removed all logging and debugging from my code
- [x] I have handled edge cases

# Screenshots

## Colors & Sizes

image

## Hover

hover

### CODE REVIEW

**Comment by zjxiaohan** [comment]
Hi Phil. I am not sure what the end result is expected to be. Could you take a look at this PR and clarify the gap between what we have right now and what the asana task is expecting?

File: src/components/Map.jsx [hunk 1]
```diff
@@ -124,50 +166,16 @@ const Map = () => {
               data={pilots}
               cluster
               clusterMaxZoom={14}
-              clusterRadius={50}
+              clusterRadius={250}
```
Comment (sphilazar) [inline src/components/Map.jsx:169]: Would prefer if this is stored as a const at the top of the file that someone on the UX team could easily recognize and change `const clusterRadius = 250`

File: src/components/Layers.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "circle-stroke-color": "#fff",
+    },
+};
+
+export const clusterLayer = {
+    id: MAP_PILOT_CLUSTERS_LAYER_ID,
+    type: "circle",
+    source: MAP_PILOTS_SOURCE_ID,
+    filter: ["has", "point_count"],
+    paint: {
```
Comment (sphilazar) [inline src/components/Layers.jsx:26]: A comment explaining how to interpret the structure of `clusterLayer.paint` would be helpful to a UX person who might edit this code to change which color displays at which zoom level (to confirm with Humanity Forward Style guide).

File: src/components/Layers.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            ],
+            100,
+            [
+                "step",
+                ["get", "point_count"],
+                20,
+                3,
+                30,
+                6,
+                40
```
Comment (sphilazar) [inline src/components/Layers.jsx:50]: Continuation of above comment - especially need clarity on these numbers

**Comment by sphilazar** [comment]
Hey Han, this is great! Intention behind this task (as I should have been more explicit about) is to enable controls for the various clustering behaviors for our UX team to be able to easily change which color appears at which zoom level (per Humanity Forward style guide etc). To make this goal even easier, would appreciate defining some `const`s and adding some comments for clarity. I love this implementation and especially like the expansion of the size of the circle on hover.

**Comment by zjxiaohan** [comment]
> Hey Han, this is great! Intention behind this task (as I should have been more explicit about) is to enable controls for the various clustering behaviors for our UX team to be able to easily change which color appears at which zoom level (per Humanity Forward style guide etc). To make this goal even easier, would appreciate defining some `const`s and adding some comments for clarity. I love this implementation and especially like the expansion of the size of the circle on hover.

@sphilazar Thank you for your reply. I added several comments and defined a const. 

I am a bit conflicted about defining more consts in `Layers.jsx` as these numbers aren't really used across files and they are pretty centralized for now. So I only added several comments. I am happy to define more if it's considered a better practice. :)

I considered creating a class called `Layer` and put all related parameters and behaviors into the class, but I found it not easy to do so considering `map` and `layer` is tightly coupled - unless we want to pass `mapRef` around (which can be pretty messy). It looks like that current level of complexity can be handled just fine. Let me know if there are better ways to organize the code.

Is there any way to know what our UI/UX team may want to edit? I searched "Humanity Forward Style" on Google and found nothing. I'm worrying that I haven't implemented all that our UX person is needed (what if they want to turn the cluster to a rainbow on hover?!).

Despite all the concerns above the pr is ready for review.

**Comment by sphilazar** [comment]
Hey Han, I think the organization of your code makes sense for now. I know Ryan will be working to add a layer that shows legislative districts, so I will rely on him to determine if we need to re-org code as needed. 

Here is our style guide / pretty much everything UX-related https://www.figma.com/file/LNvDJkTZPIcth4r9OJclRh/Basic-Income-Hub?node-id=55%3A17 (should be in the Basic Income Hub overview google doc as well)

**Review by sphilazar — APPROVED** [review]
Thanks for adding those comments!

**Comment by zjxiaohan** [comment]
> Hey Han, I think the organization of your code makes sense for now. I know Ryan will be working to add a layer that shows legislative districts, so I will rely on him to determine if we need to re-org code as needed.
> 
> Here is our style guide / pretty much everything UX-related https://www.figma.com/file/LNvDJkTZPIcth4r9OJclRh/Basic-Income-Hub?node-id=55%3A17 (should be in the Basic Income Hub overview google doc as well)

Thanks! I haven't checked the overview doc for a while.

**Pull request merged**